File: back-end/src/main/java/com/trucks/domain/valueobject/Brand.java
package com.trucks.domain.valueobject;

import java.util.Objects;

import com.trucks.utils.TruckUtils;

public class Brand {

    private static final int MAX_LENGTH = 20;
    private static final String BRAND_NULL_ERROR_MESSAGE = "Brand cannot be null or empty";
    private static final String ERROR_MAX_LENGTH =
            "Brand must be at most " + MAX_LENGTH + " characters in length";
    private final String brand;

    public Brand(String brand) {
        Brand.validate(brand);
        this.brand = brand;
    }

    public static void validate(String brand) {
        if (brand == null || brand.isEmpty()) {
            throw new IllegalArgumentException(BRAND_NULL_ERROR_MESSAGE);
        }
        if (brand.length() > MAX_LENGTH) {
            throw new IllegalArgumentException(ERROR_MAX_LENGTH);
        }
    }

    public String getBrand() {
        return brand;
    }

    public static Brand random() {
        String brand = TruckUtils.generateRandomString(TruckUtils.generateRandomNumber(1) + 1);
        return new Brand(brand);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof Brand)) {
            return false;
        }
        Brand other = (Brand) o;
        return Objects.equals(brand, other.getBrand());
    }

    @Override
    public int hashCode() {
        return Objects.hash(brand);
    }
}

File: front-end/src/domain/valueobject/brand.ts
import { Utils } from "../../utils/utils";

export class Brand {
  	
  	private readonly _brand: string;
	
  	constructor(brand: string) {
  	    Brand.validate(brand);
  	    this._brand = brand;
  	}
  
  	public static validate(brand: string): void {
		const MAX_LENGTH: number = 20;
  		const BRAND_NULL_ERROR_MESSAGE: string = "Brand cannot be null or empty";
  		const ERROR_MAX_LENGTH: string = `Brand must be at most ${MAX_LENGTH} characters in length`;
  	    if (brand == null || brand.trim() === "") {
  	        throw new Error(BRAND_NULL_ERROR_MESSAGE);
  	    }
  	    if (brand.length > MAX_LENGTH) {
  	        throw new Error(ERROR_MAX_LENGTH);
  	    }
  	}
  
  	get brand(): string {
  	    return this._brand;
  	}
  
  	public static random(): Brand {
  	    const brand: string = Utils.randomString(Utils.randomInt(0, 9) + 1);
  	    return new Brand(brand);
  	}
}
